src/mono/mono/utils/mono-mmap-wasm.c

```diff
@@ -37,22 +35,7 @@
 int
 mono_pagesize (void)
 {
-	if (mono_opt_wasm_mmap)
-		return MWPM_PAGE_SIZE;
-
-	static int saved_pagesize = 0;
-
-	if (saved_pagesize)
-		return saved_pagesize;
-
-	// Prefer sysconf () as it's signal safe.
-#if defined (HAVE_SYSCONF) && defined (_SC_PAGESIZE)
-	saved_pagesize = sysconf (_SC_PAGESIZE);
-#else
-	saved_pagesize = getpagesize ();
-#endif
-
-	return saved_pagesize;
+	return 16384;
```

Comment:
we should make sure it's actually okay to lie here

Reply:
this will improve the state of things by making pagesize not > the max block size, but the design of sgen is optimized for a base page size in the 4-8K range so that it can start with smaller blocks and then step up to larger blocks. i think with a page size of 16K it will always allocate only large blocks, and that might be bad? i'm not sure.

Reply:
We can change it down to 4K, I think

Reply:
The natural wasm page size of 64k is spoiled by the fact there is `dlmalloc` header around it anyway.
So I think we could choose arbitrary bage size and 16K is max on other platforms for sgen.

Right @BrzVlad

Reply:
@kg not sure if you are talking about other blocks. GC major blocks (for small object allocation) are at least 16k. 

I think 16k page size is just fine.

Reply:
I'm referring to the comments in the lock free allocator, which is one of the places that was consuming the page size and interacting with mmap. I guess I was under the mistaken assumption that we used the lock free allocator in more places than we do.